### PR TITLE
chore: update dev dependencies and workflow

### PR DESCRIPTION
**Pull request by crazy-max**
prep for #564

* remove hack folder
* cleanup bake def
* cleanup and update dev deps
* add eslint
* jest 27
* generate source map and licenses with ncc

in follow-up fix remaining `// eslint-disable-next-line`

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,20 +2,18 @@
...
-    "esModuleInterop": true,
-    "sourceMap": true
+    "useUnknownInCatchVariables": false,
   },
   "exclude": [
     "node_modules",
-    "**/*.test.ts"
+    "**/*.test.ts",
+    "jest.config.ts"
   ]
```
Comment (mark-dr) [inline tsconfig.json:18]: This is fine, but you could also consider switching to:

```json
"include": ["./src"]
```

File: package.json [hunk 2]
```diff
@@ -3,11 +3,11 @@
   "description": "Build and push Docker images",
   "main": "lib/main.js",
   "scripts": {
-    "build": "tsc && ncc build",
-    "format": "prettier --write **/*.ts",
-    "format-check": "prettier --check **/*.ts",
```
Comment (mark-dr) [inline package.json:8]: Why not run Prettier?

Comment (mark-dr) [reply package.json:8]: (PS, if you do run Prettier, my recommendation is to do the check step with the prettier eslint plugin)

**Review by mark-dr — CHANGES_REQUESTED** [review]

File: .prettierrc.json [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-  "printWidth": 240,
-  "tabWidth": 2,
-  "useTabs": false,
-  "semi": true,
-  "singleQuote": true,
-  "trailingComma": "none",
-  "bracketSpacing": false,
-  "arrowParens": "avoid",
-  "parser": "typescript"
-}
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply .prettierrc.json:11]: Oh didn't know about that, thanks!

File: __tests__/buildx.test.ts [hunk 4]
```diff
@@ -104,30 +116,22 @@ describe('isLocalOrTarExporter', () => {
...
-      .getExecOutput(`docker`, ['version', '--format', '{{.Server.Os}}'], {
-        ignoreReturnCode: true,
-        silent: true
-      })
-      .then(res => {
-        return !res.stdout.includes(' ') && res.exitCode == 0;
-      });
-  }
-  (isDaemonRunning() ? it : it.skip)(
+  skipItIf(!isDaemonRunning())(
```
[the remark this answers is not in the document]
Comment (mark-dr) [reply __tests__/buildx.test.ts:140]: @crazy-max you marked this as resolved, but it looks like it's still an issue in this version of the code?

[the remark this answers is not in the document]
Comment (crazy-max) [reply __tests__/buildx.test.ts:140]: As I missed that one, I will try your suggestion.

[the remark this answers is not in the document]
Comment (crazy-max) [reply __tests__/buildx.test.ts:140]: @mark-dr Doesn't seem to work:

```
 FAIL  __tests__/buildx.test.ts                                                                                                                                                                           
  ● Test suite failed to run
                                                                                                                                                                                                          
    __tests__/buildx.test.ts:140:14 - error TS1308: 'await' expressions are only allowed within async functions and at the top levels of modules.                                                         
                                                                                                                                                                                                          
    140   skipItIf(!(await isDaemonRunning()))(                                                                                                                                                           
                     ~~~~~                                                                                                                                                                                
                                                                                                                                                                                                          
      __tests__/buildx.test.ts:139:24                                                                                                                                                                     
        139 describe('getVersion', () => {
                                   ~~~~~~~
        Did you mean to mark this function as 'async'?
```

And indeed as you said it always returns true with the current impl.

[the remark this answers is not in the document]
Comment (crazy-max) [reply __tests__/buildx.test.ts:140]: Hum I don't think we can return return a promise from describe but I will check that.

[the remark this answers is not in the document]
Comment (mark-dr) [reply __tests__/buildx.test.ts:140]: I found some sample code that suggested you could, but I didn't test it. If not, the easiest solution is probably to move the check into the test itself. [this comment](https://github.com/facebook/jest/issues/7245#issuecomment-791285252) suggests you can mark a test as skipped by executing `pending()` from inside it, might be worth a try.

[the remark this answers is not in the document]
Comment (crazy-max) [reply __tests__/buildx.test.ts]: Ok afterall this condition was not necessary for this test.

**Pull request merged**